include/rocksdb/env.h

```diff
@@ -594,6 +614,17 @@ class RandomAccessFile {
     return Status::OK();
   }
 
+  // Read a bunch of blocks as described by reqs. The blocks can
+  // optionally be read in parallel. This is a synchronous call, i.e it
+  // should return after all reads have completed. The reads will be
+  // non-overlapping
+  virtual void MultiRead(std::vector<ReadRequest>* reqs) {
```

Comment:
Can we move away from std::vector? Can we use the length + pointer thing too? I am worried about the memory allocation here.

Reply:
@siying We need to anyway malloc memory for the file read, and the overhead of the actual read will be much larger.

Reply:
Interface is going to be hard to change. If we make it vector now, even if in the future we have a way to reduce the malloc, it's hard for us to get rid of it.

Reply:
Makes sense. As a policy, we can try to avoid malloc in the critical path unless absolutely required.